### PR TITLE
[CB-14901] add multiple implentations for listSdx so that the backwards

### PR DESCRIPTION
**Pull request by yg-gd**
compatiable method without includeDetatched can still be used

### CODE REVIEW

File: datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java [hunk 1]
```diff
@@ -66,6 +66,12 @@
     SdxClusterResponse create(@ValidStackNameFormat @ValidStackNameLength @PathParam("name") String name,
             @Valid SdxClusterRequest createSdxClusterRequest);
 
+    @GET
+    @Path("list")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "list SDX clusters", produces = MediaType.APPLICATION_JSON, nickname = "listSdx")
```
Comment (lajosrodek) [inline datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java:72]: Not an expert of Swagger def, but I would assume that `nickname = "listSdx"` should be something unique.

**Review by lajosrodek — COMMENTED** [review]
One note. Could you please also add UT for the controller change? :smile:

**Comment by yg-gd** [comment]
The results from the swagger compatibility check now have /sdx/list identical to how it was before the change but we've introduced a new method for those looking for all(including detached)

**Comment by yg-gd** [comment]
For reviewers consider API changes made in https://github.com/hortonworks/cloudbreak/pull/11442/files

**Review by lajosrodek — APPROVED** [review]
LGTM

**Comment by horadla23** [comment]
now this would be also a breaking change, since you are removing a parameter from an existing endpoint, right?

**Comment by horadla23** [comment]
so I am afraid, with this change, we would make a bigger mess like the one we already have, since everyone who started to use this list endpoint with flag after 2.49.0-b141 would break with this change

**Comment by horadla23** [comment]
have you checked who is affected by the breaking change of list API? based on the ticket, only PR tests in thunderhead repo is affected until now, it can be fixed I think, I would check logs if other clients are affected or not

**Comment by joshelser** [comment]
> so I am afraid, with this change, we would make a bigger mess like the one we already have, since everyone who starts to use this list endpoint with flag after 2.49.0-b141 would break with this change

> have you checked who is affected by the breaking change of list API in 2.49.0-b141?

I know that my team hasn't updated our code to account for the new parameter :)

Point taken about having to fix _something_. Either the old callers or new callers (if any). Of course, I'm selfish that I hope the old call is preserved ;)

**Review by joshelser — APPROVED** [review]
non-binding +1 from me, but this is the kind of change I was hoping to see made :)

**Comment by horadla23** [comment]
> > so I am afraid, with this change, we would make a bigger mess like the one we already have, since everyone who starts to use this list endpoint with flag after 2.49.0-b141 would break with this change
> 
> > have you checked who is affected by the breaking change of list API in 2.49.0-b141?
> 
> I know that my team hasn't updated our code to account for the new parameter :)
> 
> Point taken about having to fix _something_. Either the old callers or new callers (if any). Of course, I'm selfish that I hope the old call is preserved ;)

I am fine with this if someone, who made these changes regarding list could check if someone uses the list API with the flag, for example in CDP CLI, or somewhere else (as I requested earlier also)

> have you checked who is affected by the breaking change of list API in 2.49.0-b141? based on the ticket, only PR tests in thunderhead repo is affected until now, it can be fixed I think, I would check logs if other clients are affected or not

**Review by horadla23 — CHANGES_REQUESTED** [review]
As requested, please check in Kibana if there is any client using the list API with the new `detached` flag to check if there will be any client which can be broken after this PR is merged

You can use this filter in kibana:
`"A controller method has been intercepted" AND "SdxController.list(..)"`
If there are calls with `includeDetached` flag using `true` as a value, then those will be broken after this change possibly.

**Comment by yg-gd** [comment]
@horadla23 the logs do indicate that some have started to use this with the true value for the new parameter

Also of note here that swagger does not allow for methods to be overloaded. It will only generate one endpoint for that name

**Pull request closed without merging**